docs/source/releases/latest/945-plugin-registry-class.yaml

```diff
@@ -0,0 +1,32 @@
+refactor:
+- title: 'Plugin Registry Class Updates'
+  description: |
+    Largely refactors base.py and plugin_registry.py to consolidate all registry
```

Comment:
Maybe make a note that this makes geoips pakage a utility for other packages to use for setting up their own plugin interfaces ?  Does this need documentation at some point (I'm ok merging this PR for now without changes)